### PR TITLE
PLFM-1735: fix Evaluation/Submission migration and add integration test

### PR DESCRIPTION
**Pull request by bkng**

### CODE REVIEW

File: lib/jdomodels/src/main/java/org/sagebionetworks/evaluation/dao/SubmissionDAOImpl.java [hunk 1]
```diff
@@ -94,17 +90,14 @@
 
 	@Override
 	@Transactional(readOnly = false, propagation = Propagation.REQUIRED)
-	public String create(Submission dto) throws DatastoreException {		
+	public String create(Submission dto) throws DatastoreException {
+		EvaluationUtils.ensureNotNull(dto,  "Submission");
+		EvaluationUtils.ensureNotNull(dto, "Submission ID");
```
Comment (brucehoff) [inline lib/jdomodels/src/main/java/org/sagebionetworks/evaluation/dao/SubmissionDAOImpl.java]: should be dto.getId()

File: integration-test/src/test/java/org/sagebionetworks/IT100BackupRestoration.java [hunk 2]
```diff
@@ -609,4 +615,184 @@ public BackupRestoreStatus waitForDaemon(String daemonId) throws SynapseExceptio
...
+		// Wait for it to finish
+		status = waitForDaemon(status.getId());
+		assertEquals(DaemonStatus.COMPLETED, status.getStatus());
+		
+		// Verify restoration
+		PaginatedResults<Evaluation> evalsPaginated = synapse.getEvaluationsPaginated(0, 10);
+		assertEquals(1L, evalsPaginated.getTotalNumberOfResults());
+		Evaluation evalRestored = evalsPaginated.getResults().get(0);
+		assertNotNull(evalRestored.getEtag());
+		assertFalse("eTag was not updated on restoration", evalRestored.getEtag().equals(eval.getEtag()));
```
Comment (brucehoff) [inline integration-test/src/test/java/org/sagebionetworks/IT100BackupRestoration.java]: this seems wrong.  upon restoration, the id and etag must equal those of the original object.  this is a fundamental assumption of the stack migrator.

File: integration-test/src/test/java/org/sagebionetworks/IT100BackupRestoration.java [hunk 3]
```diff
@@ -609,4 +615,184 @@ public BackupRestoreStatus waitForDaemon(String daemonId) throws SynapseExceptio
...
+		assertEquals(DaemonStatus.COMPLETED, status.getStatus());
+		
+		// Verify restoration
+		PaginatedResults<Submission> subsPaginated = synapse.getAllSubmissions(eval.getId(), 10, 0);
+		assertEquals(1L, subsPaginated.getTotalNumberOfResults());
+		Submission submissionRestored = subsPaginated.getResults().get(0);
+		assertEquals(submission, submissionRestored);
+		SubmissionStatus submissionStatus = synapse.getSubmissionStatus(submission.getId());
+		assertEquals(submission.getId(), submissionStatus.getId());
+
```
Comment (brucehoff) [inline integration-test/src/test/java/org/sagebionetworks/IT100BackupRestoration.java]: check that the restored submission status is equal to the original submission status

Comment (bkng) [inline integration-test/src/test/java/org/sagebionetworks/IT100BackupRestoration.java]: Hi Bruce -

I've made the changes we discussed earlier today, and have committed them
to pull 307. Could you please have a look and merge this request when
convenient?

Thanks,
Bennett

**Pull request merged**